### ISSUE TITLE
[Intel 10th Gen - Intel Iris Plus Graphics] Chrome OS not booting

### ISSUE THREAD

**Issue by duke7553** [sebanc/brunch#18]
When booting brunch, it went through all the steps for mounting the filesystem, but has stayed put at these last few lines for nearly 15 minutes.

**Comment by duke7553** [comment]
![IMG_20200321_121155](https://user-images.githubusercontent.com/20365014/77230827-7cf10900-6b6d-11ea-863f-421d7ab5f53d.jpg)

**Comment by duke7553** [comment]
This device is a Surface Laptop 3

**Comment by duke7553** [comment]
Update: Trying to switch to a tty with CTRL ALT Fx gave me this output:
![IMG_20200321_121659](https://user-images.githubusercontent.com/20365014/77230897-fab51480-6b6d-11ea-8f72-34e0ab971a17.jpg)

**Comment by sebanc** [comment]
Hi,

There is clearly an issue here, could you try to press e when you see briefly the grub menu at startup, add "module_blacklist=intel-ipts" after cros_debug and then press F10 to boot ?

If it does not work, could you attach a short video of the full boot process ?

**Comment by duke7553** [comment]
Will do right now. Thanks

**Comment by duke7553** [comment]
@sebanc https://www.youtube.com/watch?v=HcK6jTm-bB4

**Comment by sebanc** [comment]
Which recovery image are you using ?

**Comment by duke7553** [comment]
@sebanc I'm using rammus with the newest brunch package.

**Comment by sebanc** [comment]
Could you try adding "module_blacklist=intel-ipts i915.alpha_support=1" to the kernel command line
and upload an other video if it does not boot ?

**Comment by duke7553** [comment]
@sebanc Sure, I'll get to it

**Comment by duke7553** [comment]
@sebanc Adding those to the kernel loads the boot output for a few seconds then boots me into a black, backlit screen that flickers off every now and then.

**Comment by sebanc** [comment]
I would say that your (very recent) intel gpu is not supported in kernel 4.19. I intend to build kernel 5.4 at some point which should be compatible with your device but for now i don't have an alternative solution.

I will update this issue when kernel 5.4 is available.

**Comment by duke7553** [comment]
I tried copying a 5.4 kernel onto the USB with its modules, but it didn't quite work because I got a black screen after grub.

**Comment by sebanc** [comment]
Hi, I have not moved to kernel 5.4 yet but I fixed an issue which might have prevented your laptop from booting. If you are still interested, you can give the new testing release a try.

**Comment by duke7553** [comment]
You bet I'll give this a try! 😀

**Comment by duke7553** [comment]
@sebanc That new release didn't seem to fix it

**Comment by duke7553** [comment]
@sebanc I think v5.4 might be more compatible because I tried that copying that one to another Chromium OS spin and was able to boot to a live USB. It didn't boot after installing the OS though.

**Comment by sebanc** [comment]
Hi,

Just updating this issue to say that moving to kernel 5.4 seems to be more complicated than expected as it introduces several issues and I don't have the time to maintain 2 different kernels currently.

Also, contrarily to what I first thought, some 10th gen intel are correctly booting brunch so I think this issue is just related to a specific patch that has to be backported to kernel 4.19.

Would you be interested in trying builds to hopefully solve the issue with kernel 4.19 ?

Another possibility would be for you to build kitakar's kernel and i can explain how to add it to brunch.

**Comment by duke7553** [comment]
@sebanc Sure! I'd be happy to rapid-fire test builds for you.

**Comment by duke7553** [comment]
@sebanc kitakar actually publishes binaries for the kernel, can I add those files to brunch?

**Comment by sebanc** [comment]
Nice, let's try to fix this !

1st part so that you are not left out anymore, how to install kitakar's kernel:
- Create a brunch usb,
- Mount partition 7 of the brunch usb,
- Extract kitakar's kernel package somewhere,
- Copy /boot/vmlinuz from kitakar's kernel package to <brunch_partition_7>/kernel
- Extract modules.tar.gz into the directory <brunch_partition_7>/lib/
- Unmount partition 7 of the brunch usb.
- That should be it, try booting the usb

Limitations:
- Use 'eve' recovery if you want to be able to update chromeos
- Don't update brunch framework as it would replace the kernel (that should not be necessary for a while)
- Maybe some others i am not thinking of right now...

**Comment by sebanc** [comment]
Regarding brunch kernel 4.19, I will build a new test release with a patch I have spotted.

**Comment by LFmove** [comment]
Waiting for test the next release on my Surface Pro 7! 💪 
Thanks

**Comment by sebanc** [comment]
@LFmove @duke7553 I have added you to a private repo on my git named "brunch-testing" that I use for issues that are difficult to debug and might break things for others.

When you have the time, could you try the release that's there ?

Very important: you have to add to the kernel command line "options=iwlwifi_backport", 2 possible ways to do it:
- Method1: Mount partition 12 of your usb in linux and add "options=iwlwifi_backport" after "cros_debug" in /efi/boot/grub.cfg.
- Method2: press "e" quickly at boot when you see grub and add "options=iwlwifi_backport" after "cros_debug" in /efi/boot/grub.cfg, then press F10 or CTRL-X to boot.

**Comment by LFmove** [comment]
Tried,

Same thing (with the method 2), It stops.

Attached picture with the issue.
![IMG_20200511_184650](https://user-images.githubusercontent.com/65065497/81592903-f2839380-93be-11ea-8d6c-77658d9a9cd3.jpg)



Available for other tests. 👍

**Comment by sebanc** [comment]
Thanks ! Could you try booting with "module_blacklist=intel_ishtp_hid" ?

**Comment by LFmove** [comment]
Sorry, i read it now.

Add the command with the others, or just this?

**Comment by sebanc** [comment]
No worries,

The best would be to add "module_blacklist=intel_ishtp_hid options=iwlwifi_backport"

**Comment by LFmove** [comment]
Ok, 10min and i try with :

Method2: press "e" quickly at boot when you see grub and add:

 options=iwlwifi_backport 

 cros_debug

/efi/boot/grub.cfg 

module_blacklist=intel_ishtp_hid options=iwlwifi_backport

then press F10 or CTRL-X to boot.

- Right?

**Comment by sebanc** [comment]
Not sure:

Press "e" quickly at boot when you see grub:
1) locate the "cros_debug" text on the screen
2) add "module_blacklist=intel_ishtp_hid options=iwlwifi_backport" after the "cros_debug" text
3) press F10 or CTRL-X to boot.

**Comment by LFmove** [comment]
Same error. Stops always at the same line (HDA Intel PCH HDMI/DP

But i see this lines (attached pic)
![IMG_20200512_101512](https://user-images.githubusercontent.com/65065497/81659308-e1c43380-9439-11ea-8eb8-f0da503baadb.jpg)

**Comment by sebanc** [comment]
I doubt it could be the audio driver but we never know, could you try booting with "module_blacklist=snd_hda_intel options=iwlwifi_backport" ?

**Comment by sebanc** [comment]
Forget my previous post, I just saw mostly the same log from someone who tried to boot chromeos with "nomodeset" which most likely means that this issue is related to the i915 driver not loading at all.

What's weird is that intel 10th gen laptops were able to boot brunch on kernel 4.19 but I guess these have a different gpu model.

I will try to see if I can find documentation on the SP7/SL3 gpu and to figure out the best way to move forward with this.

Meanwhile you can try kitakar's kernel.

**Comment by LFmove** [comment]
Ok ok than you! Sorry but Yesterday i really can't try.

**Comment by sebanc** [comment]
No worries, just try things when you have the time, I just wrote this because I saw the same log errors and the issue became much clearer.

I am currently backporting a few patches I found for i915 10th gen and will probably upload a new build that you can try whenever you have the time (no rush).

**Comment by sebanc** [comment]
I have uploaded a new test release in the "brunch-testing" repository, could you try it with "options=iwlwifi_backport" added to the kernel command line when you have the time ?

**Comment by LFmove** [comment]
Sure. ( Always after the "cros_debug" text, etc...?)

**Comment by LFmove** [comment]
Sure. ( Always after the "cros_debug" text, etc...?)

**Comment by sebanc** [comment]
yes, just "options=iwlwifi_backport" after the "cros_debug" text

**Comment by LFmove** [comment]
Ok, done It!

It's like finished, but when It reboot, you can ear a sound like ChromeOs starting, but the screen star flashing, "dark, grey, dark grey" (never stops)

Anyway...
I see some error in the Building operation. (Pic)
![IMG_20200513_192631](https://user-images.githubusercontent.com/65065497/81844952-280bb680-9550-11ea-9922-ab556586ca40.jpg)

**Comment by LFmove** [comment]
However, it is the first time that "builing" ends and the PC is restarted. (cool for me)

**Comment by thorstenludwigs** [comment]
I did some testing with my surface Pro7, the problem I had that it went to a black flickering screen ("i915.alpha_support=1"). If I connect, an external Monitor, I get a blank white page on it. 

I tested kitakar's latest 5.4 kernel. It works, but I have Wifi and Bluetooth issues. 
Hope this helps a bit and thanks for your effort!

**Comment by sebanc** [comment]
@LFmove The error while rebuilding rootfs is harmless, it's just an issue with the patch ordering when using "options=iwlwifi_backport", I will fix it.

@thorstenludwigs Thanks for the report, indeed it seems the problem with 4.19 kernel seems to be related to the internal display. I tried to identify the commit which fixes it in the i915 driver but there are too many.

For now, I will try to build an experimental 5.4 kernel that you can install manually.

**Comment by thorstenludwigs** [comment]
Tanks for your help. I will compile a test kernel tomorrow as well.

**Comment by kitakar5525** [comment]
Leaving note here that making SP7/SL3 wifi work with (my prebuilt) 5.4 kernel is being discussed also at here: https://github.com/kitakar5525/chromeos-kernel-linux-surface/issues/17

**Comment by LFmove** [comment]
News for our Surface? 🙂

**Comment by duke7553** [comment]
@sebanc Can you provide an update on v5.4 kernel?

**Comment by sebanc** [comment]
Hi,

Both @kitakar5525 and I are working on 5.4 kernel with what seems to be relatively similar issues for now.

I will post my next bootable build here so that at least you have something to try.

**Comment by sebanc** [comment]
Hi,

I have added everyone in this issue to a new private "brunch-testing" branch and uploaded a 5.4 build which appears to boot (at least on my Surface Go).

https://github.com/sebanc/brunch-next/releases

As previously mentioned, it's difficult to find the time to maintain 2 different kernels so I cannot guarantee regular/timely updates.

**Comment by kitakar5525** [comment]
I also tried the 5.4 build (`brunch_r83_k5.4_testing_20200604.tar.gz`) and it's working on SB1 (NVMe disk).

So, I hope it also works on SP7/SL3. Regarding wifi, I heard it works with Linux 5.4's upstream iwlwifi (`CONFIG_IWLWIFI`) and that build uses this one, so hopefully wifi also works here.

@sebanc I'm curious if we can get SP7/SL3 wifi working with chromiumos' iwlwifi (`CONFIG_IWL7000`). I thought it's new enough but I heard it's unstable.
Do you know anything about the iwl7000?

By the way, it doesn't boot on Surface 3, saying that "brunch: ChromeOS was not found on any device, falling back to shell..."
This is maybe because mmc_block is built as a module (`CONFIG_MMC_BLOCK=m`). All of S3 models use eMMC.

**Comment by sebanc** [comment]
Thanks for testing this release :)

Regarding CONFIG_IWL7000, it seems to be an alternative driver which means to have full-MAC support (not sure what that really means):
https://chromium.googlesource.com/chromiumos/third_party/kernel/+/2ce11a42c53960e9d40cd95dbf53ed0bdfe2bd4e
I enabled it at some point but it created issues for nearly all iwlwifi users, maybe it requires specific firmwares to work properly...

For the S3, nice catch ! I will correct it in a future build.

**Comment by duke7553** [comment]
@sebanc Booting the brunch v5.4 image you published, it appears that WiFi does not work at all on my Surface Laptop 3.

**Comment by sebanc** [comment]
@duke7553 that's weird, could you attach a dmesg (run "dmesg > ~/Downloads/dmesg.txt" and attach dmesg.txt to this issue.

Then, you can try booting with "options=iwlwifi_backport" added to the kernel command line and see if it helps.

**Comment by duke7553** [comment]
@sebanc Sorry! It works now after installing it to the internal drive. Great work!

**Comment by kitakar5525** [comment]
@sebanc 
> Regarding CONFIG_IWL7000,
> [...]
> I enabled it at some point but it created issues for nearly all iwlwifi users, maybe it requires specific firmwares to work properly...

I see... Thank you for the info.

**Comment by Warelik** [comment]
Hi. Any update about this?

**Comment by sebanc** [comment]
@Warelik For now you have to use the specific release here:
https://github.com/sebanc/brunch-next/releases

**Comment by Warelik** [comment]
[lshw.pdf](https://github.com/sebanc/brunch/files/4781262/lshw.pdf)
Feels like everything works. The camera surprisingly works in crouton but does not work in chrome. There are some oddities in the work of android applications, but nothing critical so far.
Thank's @sebanc

**Comment by sebanc** [comment]
@Warelik Thanks for the feedback :) There is currently some camera compatibility issues with rammus image, you can try to see if the camera works on "reef" which has the previous usb camera framework and should be more compatible.

**Comment by weikevinhc** [comment]
Sorry to be a bother but can i get the invite to the brunch-next release again?

**Comment by sebanc** [comment]
@weikevinhc validate the github invite email and you should have access now.

**Comment by LFmove** [comment]
Trying on Surface pro7, i'm in ChromeOS finally, but i try the command : mkdir -p ~/tmpmount sudo mount dev/sda5 ~/tmpmount

and it say 

mkdir: cannot create directory ‘sudo’: Permission denied
mkdir: cannot create directory ‘mount’: Permission denied
mkdir: cannot create directory ‘/dev/sda5/’: File exists


can you help me?

**Comment by weikevinhc** [comment]
So I managed to boot into chromeOS using brunch_r83_k5.4_testing_20200604. The wifi was initially not working but after adding "options=iwlwifi_backport", the wifi appears to be working. However, although the wifi is constantly connected, it would go back and forth from connected with no internet to connected. I cannot sign into my google account when prompted because the internet connection would be lost. Using guest mode to bypass the sign in, i can get some pages to load in chrome.

I also have an unrelated questions (perhaps i should start another thread). When i generate chromeos.img with chromeos-install.sh, regardless of the value i give to -s, when I boot into chromeOS, it always shows as having ~7Gb of available space. Am i doing something wrong?

**Comment by Warelik** [comment]
> @Warelik Thanks for the feedback :) There is currently some camera compatibility issues with rammus image, you can try to see if the camera works on "reef" which has the previous usb camera framework and should be more compatible.

Now it looks like everything is working :) Thanks

**Comment by LFmove** [comment]
Tested on a Surface PRO7, everything works fine, including wifi. The only two things that don't work best are: the touchscreen, it works but I have to press more on the screen, and the microsoft type cover, which works perfectly but when I unplug it and re-connect it takes 5-10sec to change from tablet to keyboard mode.
However it is really a great job! I can't wait for the "stable" version to come out !!!!

**Comment by LFmove** [comment]
Something wrong also with bluetooth. Doesn't work well

**Comment by akihitoh** [comment]
Could you invite the brunch-next release? My One Mix 3 Pro has also the same issue and would like to test the brunch-next.

**Comment by sebanc** [comment]
@LFmove Thanks for the feedback, I will try to see if there is an updted Ipts module from StollD on the linux-surface repo.

@akihitoh You should have access to the brunch-next repo now.

**Comment by akihitoh** [comment]
@sebanc Thank you for your invitation. I do not see any problems on a One Mix 3 Pro now. I will try the next for other UMPCs.

**Comment by DavidIlie** [comment]
Hi! I recently bought a HP Spectre 2019 with a 10th gen CPU and I want to get brunch working. I am at the same error in the images at the very beginning. Is there a chance I could be added to the repo with the 5.4 kernel?

Thanks!

**Comment by sebanc** [comment]
@DavidIlie You have been added, validate the email invite and the release is here:
https://github.com/sebanc/brunch-next/releases

**Comment by DavidIlie** [comment]
Thanks!

**Comment by DavidIlie** [comment]
Do you have any recommendations for the flags that I should add to grub?

**Comment by DavidIlie** [comment]
Hello! I booted chromeOS with no extra boot flags using the reef image and everything works fine! The only issue is that Wi-Fi speeds are a little slow and the coverage is not so good. Is there a boot flag that can fix this? 

Thanks!

**Comment by sebanc** [comment]
@DavidIlie What's your wireless driver ? (if you have doubts, run "lsmod > ~/Downloads/lsmod.txt" in crosh shell and attach lsmod.txt from your downloads folder to this issue)

**Comment by DavidIlie** [comment]
Here it is.
[lsmod.txt](https://github.com/sebanc/brunch/files/4807996/lsmod.txt)

**Comment by DavidIlie** [comment]
It is a Wi-Fi 6 card but has perfect compatibility in ubuntu for example

**Comment by DavidIlie** [comment]
Also I have just discovered that sound does not work

**Comment by sebanc** [comment]
Could you try to add "options=iwlwifi_backport module_blacklist=snd_hda_intel" to the kernel command line ?

**Comment by DavidIlie** [comment]
ok

**Comment by DavidIlie** [comment]
The Wi-Fi has a reasonable speed but websites load very slow. Also sound and shutdown still doesn't work

**Comment by sebanc** [comment]
Try "options=iwlwifi_backport module_blacklist=snd_sof"

**Comment by DavidIlie** [comment]
The connection is a little better but the speeds are still low. I have tried again with no kernel arguments and Wi-Fi now works perfectly so I don't think Wi-Fi is a issue. However, sound and shutdown are two really important things :)

**Comment by DavidIlie** [comment]
Perhaps I should try rammus instead of reef?

**Comment by sebanc** [comment]
Ok, for sound I will need 2 dmesg:
- One with "module_blacklist=snd_hda_intel"
- One with "module_blacklist=snd_sof"

For shutdown, I have no idea right now, usually it works out of the box.

**Comment by DavidIlie** [comment]
Alright

**Comment by DavidIlie** [comment]
dmesg for snd_hda_intele
[dmesg_snd_hda_intel.txt](https://github.com/sebanc/brunch/files/4808123/dmesg_snd_hda_intel.txt)

**Comment by DavidIlie** [comment]
dmesg for snd_sof
[dmesg_snd_sof.txt](https://github.com/sebanc/brunch/files/4808133/dmesg_snd_sof.txt)

**Comment by sebanc** [comment]
Could you extract the below archive in /lib/firmware/intel/, keep "module_blacklist=snd_hda_intel" in the kernel command line and attach another dmesg ?
[sof.zip](https://github.com/sebanc/brunch/files/4808149/sof.zip)

**Comment by DavidIlie** [comment]
Here it is
[dmesg_new.txt](https://github.com/sebanc/brunch/files/4808175/dmesg_new.txt)

**Comment by sebanc** [comment]
Could you try adding "options=iwlwifi_backport module_blacklist=snd_sof snd_hda_intel.dmic_detect=0" to the kernel command line ?

**Comment by sebanc** [comment]
For shutdown, you can also try adding "intel_iommu=off" to the kernel command line.

**Comment by DavidIlie** [comment]
I have put every kernel argument apart from options=iwlwifi_backport because it made my internet speed slow. Now everything works! Sound (a little quiet but meh), Wi-Fi and shutdown (even though it takes 1 minute). Thank you very much!

**Comment by akihitoh** [comment]
I have tested tht next to GPD Pocket 1 since kernel 5.4 includes several patches for GPD Pocket. However, I got the following message. I guess that Atom x7-Z8750(Cherry Trail) is not appropriate. 

> brunch: ChromeOS was not found on any device

The GPD Pocket can be booted by the image that is deployed to an external disk and initialized by an another PC. I think that it is not recommented. The touchscreen, input device, sound and wifi work well. I am looking forword that the next will support GPD Pocket in a further version.

**Comment by DavidIlie** [comment]
Hi! this version of brunch was running so well with chromeos that I decided to install! It was running fine untill I realised that the microphone is not working. Is there any way to fix it?

**Comment by sebanc** [comment]
@DavidIlie Unfortunately microphone is a quite common issue that I have not been able to fix but some have found workaround methods in issue #56.

**Comment by sebanc** [comment]
Hi, I have just uploaded a new test release with 5.4 kernel. It mostly fixes the boot from emmc issue (for Surface 3 and most probably GPD Pocket).

@DavidIlie You should not need "blacklist=snd_sof snd_hda_intel.dmic_detect=0" kernel parameters anymore but microphone will most likely still not work.

**Comment by DavidIlie** [comment]
How would I update brunch if I installed? Do I still do the same method as with the stable one?

**Comment by sebanc** [comment]
You can follow those steps:
https://github.com/sebanc/brunch#update-only-the-brunch-framework-if-you-have-enabled-native-chromeos-updates

**Comment by DavidIlie** [comment]
I have just updated brunch and sound does not work without the parameters. I had to put them back in.

**Comment by sebanc** [comment]
> I have just updated brunch and sound does not work without the parameters. I had to put them back in.

That's interesting, could you post a dmesg when you have the time ?

**Comment by DavidIlie** [comment]
[dmesg.txt](https://github.com/sebanc/brunch/files/4836886/dmesg.txt)
Here it is

**Comment by DavidIlie** [comment]
Here is the correct one.
[newdmesg.txt](https://github.com/sebanc/brunch/files/4836900/newdmesg.txt)

**Comment by derqami** [comment]
Hi, I want to try chromeos on onemix pro 3. Is it possible you (sebanc, I think) send me invitation for test build K 5.4 ?. Sorry for interrupting.

**Comment by sebanc** [comment]
@derqami You have been added to "brunch-next", validate the github invite email and the release is here:
https://github.com/sebanc/brunch-next/releases

**Comment by akihitoh** [comment]
@sebanc Thank you for the update. The current version work well in my GPD Pocket.

**Comment by derqami** [comment]
Thank you very much for such a great work. I used latest k5.4 ,multi_install.sh and rammus recovery...Everything works quite fine on onemix pro 3. Stylus and accelerometer do not work which is not a big deal for me.

**Comment by silatiol** [comment]
Hi, I want to try chromeos on my new xps 13. Is it possible to get added to the "brunch-next" repo?

Thank you @sebanc

**Comment by sebanc** [comment]
@silatiol you have been added to the "brunch-next" repo

**Comment by Andy4Shurr** [comment]
@sebanc I'm trying to get ChromeOS working on the MiPad 2 (atom x5 z8500). Please could I test the latest kernal from the "brunch-next" repo as well, seeing as it's a similar SOC to the the GPD?

**Comment by sebanc** [comment]
@Andy4Shurr You have been added to the brunch-next repo, however your device should theoretically work fine with kernel 4.19 and no issue reports are accepted on 5.4 kernel aside from 10th gen intel devices.

**Comment by kar200** [comment]
Hi @sebanc , I am not sure this is the correct way but I am using a matebook 13 (2020) with intel 10th generation CPU .

My wifi is not starting. I am using latest rammus from today on triple boot (Windows, Ubuntu, ChromeOS). I used a script "multi_install.sh" and it seems to be working "except my wifi".

I will be happy to test the 5.4 kernel if its possible.

Thanks again for your work I couldn't believe this was possible (took a long time for me to choose between the matebook and a chromebook when it can do both).

**Comment by loopy-git** [comment]
@sebanc Could I get access to the brunch-next repo?
Thanks!

**Comment by qtotter** [comment]
Hi @sebanc , I'm qtotter from discord. Could you kindly provide me access to 5.4 branch, please? Thanks in advance!

**Comment by sebanc** [comment]
Hi @qtotter, you should have received a GitHub invite.

**Comment by qtotter** [comment]
Hi @sebanc, Thanks! I am learning about Brunch to get ready now by the time my new PC arrives.

**Comment by Namaneo** [comment]
Hi all,

Finally got a working install on the SSD of my XPS 13 (@sebanc thank you for the invitation on the 5.4 repo !). Basically everything seems to work properly except the wifi: same symptoms as others here, the connection is up but extremely slow/unstable. 

After searching a little, I found an article talking about network issues related to power management of the adapter, and the ping issue he shows seems similar (at least in my case): 
https://rolandschnabel.de/blog/2019/05/slow-wifi-network-on-linux-laptop/ .

Unfortunatly, I don't think the workaround can be apply here, and I don't have the knowledge to go deeper, @sebanc do you think our issues could be related? Is there something we can try?

Thanks for the great work by the way, it's amazing!

**Comment by chenshen2830569316** [comment]
I have a Surface Pro 7（inter i7 1065G7）and I am happy to help the test.

**Comment by chenshen2830569316** [comment]
I have a Surface Pro 7（intel i7 1065G7）and I am happy to help the test.

**Comment by guangleijia86** [comment]
Hi @sebanc! I have a Surface pro 7 with intel 1035G4, and I'd appreciate if you could add me to the brunch next repo. Thanks a lot.

**Comment by sebanc** [comment]
@chenshen2830569316 @guangleijia86 you should have received the github invite to access brunch-next repo

**Comment by chenshen2830569316** [comment]
@sebanc  Will I think I haven’t received the invitation, maybe I don’t know where to agree.

**Comment by chenshen2830569316** [comment]
I'm so sorry to bother you.Now I had already join in this project.

**Comment by chenshen2830569316** [comment]
![56A6230A-9897-4E30-9DD6-E4C0093153CF](https://user-images.githubusercontent.com/61828328/91312568-f65ce980-e7e6-11ea-949c-0dd064d4fc85.jpeg)
Thanks a lot.

**Comment by guangleijia86** [comment]
Hello! @sebanc I just install chromeos in my surface  but it  can’t boot chromeos what should I do? 
![CB2E6899-2ACE-43F0-AD86-93A142FF22BD](https://user-images.githubusercontent.com/70249402/91372774-26d47000-e847-11ea-81dc-455e7b45cf05.jpeg)

**Comment by princepjamie06832** [comment]
I have my laptop with 10th Gen Intel Core i3-1005G1. You can add me for brunch next-repo?

**Comment by sebanc** [comment]
@princepjamie06832 I added you to brunch next. If you have issues with 5.4, I think i3 (only) 10th gen should also work with kernel 4.19

**Comment by princepjamie06832** [comment]
> @princepjamie06832 I added you to brunch next. If you have issues with 5.4, I think i3 (only) 10th gen should also work with kernel 4.19

I got your invitation and working okay with kernel 5.4, but Play Store app isn't working. I have ASUS VivoBook X512JA (10th Gen Intel Core) and running on rammus. All drivers is working for now. I had issues with kernel 4.19, it won't boot correctly on 10th Gen Intel Core.

**Comment by sebanc** [comment]
@princepjamie06832 You can try to reset google sync: https://support.google.com/chromebook/answer/2914794?hl=en

**Comment by princepjamie06832** [comment]
> 
> 
> @princepjamie06832 You can try to reset google sync: https://support.google.com/chromebook/answer/2914794?hl=en

I tried this, but didn't worked. It says "Something went wrong" when taking too long to setting up Play Store app. I might wait further.

**Comment by byransays** [comment]
Hello. Just got a new Surface Pro 7. Would be more than happy to test.

**Comment by chenshen2830569316** [comment]
Well,I think there are something wrong in my brunch(k5.4). I can't use google play store.

**Comment by princepjamie06832** [comment]
> Well,I think there are something wrong in my brunch(k5.4). I can't use google play store.
> 
We tried to reset Google Sync, but no luck. Upcoming framework will fix this issue that Play Store takes too long to set up. I tried to use "android_init_fix", but nothing worked. @sebanc is working on it.

**Comment by PrestonCrowe** [comment]
Hello! I just got an Acer spin with a 10th gen processor, and I would love to test for you!

**Comment by sebanc** [comment]
@PrestonCrowe You should have received an invite for brunch-next repo.

@princepjamie06832 @chenshen2830569316 Could you confirm if what break android apps is:
Updating to ChromeOS r85 (even with the older brunch version) or updating brunch to r85 ?

**Comment by chenshen2830569316** [comment]
> @PrestonCrowe You should have received an invite for brunch-next repo.
> 
> @princepjamie06832 @chenshen2830569316 Could you confirm what breaks android apps:
> Updating to ChromeOS r85 (even with the older brunch version) or updating brunch to r85 ?

I just use samsung galaxy chromebook recevory bin,(chromeos_13099.110.0_hatch_recovery_stable-channel_mp-v6.bin)

**Comment by chenshen2830569316** [comment]
And I can't see the Setup complete page. I don't know is it related to this( I waited about six hours,but it still display please wait）

**Comment by chenshen2830569316** [comment]
I don't have a chromebook so I can't provide more help about this question.

**Comment by princepjamie06832** [comment]
> @PrestonCrowe You should have received an invite for brunch-next repo.
> 
> @princepjamie06832 @chenshen2830569316 Could you confirm what breaks android apps:
> Updating to ChromeOS r85 (even with the older brunch version) or updating brunch to r85 ?

Nope that didn't worked. Same problem when I tried to use older version and updating latest framework and Chrome OS.

**Comment by sebanc** [comment]
@princepjamie06832 If you install the old release and update only chromeos (not brunch), are android apps working ?

**Comment by PrestonCrowe** [comment]
Thanks for the invite @sebanc! I got it running on my laptop with only a few issues. Touch and pen don't work yet, the touchpad doesn't work yet, no microphone, and android errors out like the others. But otherwise, it works great!

**Comment by princepjamie06832** [comment]
> @princepjamie06832 If you install the previous brunch release and update only chromeos (not brunch), are android apps working ?

Android app doesn't work on latest OS with older version of framework. I tried to use "android_init_fix" option, reset Google Sync and updating latest OS on older brunch, but nothing worked. Currently, I'm running on hatch recovery for 10th Gen.

**Comment by sebanc** [comment]
@princepjamie06832 Thanks for the feedback, could you try with rammus recovery image ? I will test it again but normally android apps are part of my "test" plan and unless I missed something it worked on r85 with rammus recovery image on my Surface Go.

@PrestonCrowe Not sure if it is the same issue but there are some tips to help with acer touchpad/touchscreen here:
https://community.acer.com/en/discussion/604971/spin-5-sp513-54n-nx-hquec-003-touchscreen-stylus-touchpad-and-installation-issues-on-ubuntu-20-04

**Comment by PrestonCrowe** [comment]
Thanks for the tip. I got the touchscreen working and stylus partially working. The touchpad works in crouton now, just not normal Chrome OS, but I haven't worked step 2 of the touchpad rules yet. (The tip also helped me with some issues in my Hackintosh as well!)

**Comment by chenshen2830569316** [comment]
> > @princepjamie06832 If you install the previous brunch release and update only chromeos (not brunch), are android apps working ?
> 
> Android app doesn't work on latest OS with older version of framework. I tried to use "android_init_fix" option, reset Google Sync and updating latest OS on older brunch, but nothing worked. Currently, I'm running on hatch recovery for 10th Gen.

I just try install r83 recevory image on my surface pro 7,but google play store didn't work.

**Comment by qtotter** [comment]
@chenshen2830569316 , which Brunch are you using with r83?

**Comment by chenshen2830569316** [comment]
> @chenshen2830569316 , which Brunch are you using with r83?

brunch_r85_k5.4_testing_20200905

**Comment by charliesbot** [comment]
@sebanc I'm facing the same issues, using an xps 13 9300. Could you send me an invitation too? I would love to test it using my laptop!

**Comment by sebanc** [comment]
@charliesbot You should have received an invite, maybe try r83 it seems there is an issue with android apps in current r85

**Comment by sebanc** [comment]
I have not had the time to reinstall the previous r85 to verify if android apps worked for me but I have just added a new r85 build that I tested with android apps working on both chromeos r84 and r85. Could you confirm if you still experience the android apps not working issue in this build ?

**Comment by chenshen2830569316** [comment]
> 我没有时间重新安装以前的r85来验证Android应用程序是否为我工作，但我刚刚添加了一个新的r85版本，我测试了Android应用程序工作在铬 r84 和r85。你能确认，如果你仍然体验Android应用程序不工作的问题，在这个版本？

I will try right now.

**Comment by chenshen2830569316** [comment]
failed
![C1391365-14B5-4741-8B3D-1E7977C1DD03](https://user-images.githubusercontent.com/61828328/93360858-c13c3800-f876-11ea-8e45-e569333dd426.jpeg)

**Comment by chenshen2830569316** [comment]
I used a vpn to activation it but it didn’t work,the problem same as before.

**Comment by chenshen2830569316** [comment]
It is not a network reason, if it is a network problem, other notifications will be displayed.

**Comment by princepjamie06832** [comment]
> It is not a network reason, if it is a network problem, other notifications will be displayed.

Same issue after testing released yesterday.

**Comment by princepjamie06832** [comment]
Same problem. I couldn't get Android app to work.

**Comment by sebanc** [comment]
@chenshen2830569316 @princepjamie06832 Do android apps work for you with the brunch r83 release ?

**Comment by chenshen2830569316** [comment]
> @chenshen2830569316 @princepjamie06832 Do android apps work for you with the brunch r83 release ?

Will try right now

**Comment by chenshen2830569316** [comment]
failed

**Comment by chenshen2830569316** [comment]
failed
![62AF24FA-27C2-4E4A-8A9F-5F30F034A0C3](https://user-images.githubusercontent.com/61828328/93495668-f2813a80-f940-11ea-8b9f-3f5922b892c3.jpeg)

**Comment by chenshen2830569316** [comment]
The same problem like yesterday

**Comment by chenshen2830569316** [comment]
There are some another problem in r85,when I tried to close it, the system restartd when I closed it successfully.

**Comment by chenshen2830569316** [comment]
There are some another problem in r83,when I tried to close it, the system restartd when I closed it successfully.

**Comment by princepjamie06832** [comment]
> There are some another problem in r83,when I tried to close it, the system restartd when I closed it successfully.

What framework version running on?

**Comment by chenshen2830569316** [comment]
> > r83中还有另一个问题，当我尝试关闭它时，成功关闭它后系统会重新启动。
> 
> 运行什么框架版本？

I think it might be my SDcard problem,because I can’t use it after I shut down my surface.

**Comment by sebanc** [comment]
@chenshen2830569316 @princepjamie06832 try installing brunch to the hdd with the dual boot method, some sdcard / usb are too slow for android apps and crostini to load.

Also @chenshen2830569316 brunch with kernel 5.4 has been reported to work on SP7 by several users so I don't think your issue is actually related to brunch.

**Comment by chenshen2830569316** [comment]
> @chenshen2830569316@princepjamie06832安装早午餐到高清与双引导方法， 一些 sdcard / usb 太慢了， Android 应用程序和 crostini 加载。
> 
> 此外@chenshen2830569316 5.4 的早午餐已被几个用户报告在 SP7 上工作， 所以我不认为你的问题实际上与早午餐有关。

It may be related to the version I chose，I will try another version later,thank you.

**Comment by bogdanenedoghitoiu** [comment]
Hi, I tried installing ChromeOS on a Lenovo IdeaPad 5 14IIL05 and I get the exact same error as the one in the screenshots above.  I used rammus, hatch and eve images of various versions.

 Added a comment here as advised on the discord server.

**Comment by sebanc** [comment]
@bogdanenedoghitoiu you should have received the brunch-next invite

**Comment by bogdanenedoghitoiu** [comment]
@sebanc I got it, thank you so much. I will try to install again today and I will come back with updates.

**Comment by bogdanenedoghitoiu** [comment]
update: I managed to successfully install ChromeOS using the brunch test script and rammus v85. The play store is not working but I don't mind it since I want to use this laptop mainly as a development environment. 

I do have a question though, do I need access to play store in order to have the ADB server (for testing and debugging android/flutter apps from android studio) up and running? At the moment i can connect to it after enabling the adb debugger from settings but the device appears as offline.

**Comment by qtotter** [comment]
While I'm still waiting for my PC to be delivered, I've been reading all the comments here, but what I don't understand is whether you cannot use Play Store even with any versions of Brunch next, or is it a specific combination of Brunch next and Chrome image, or a specific combination of CPU/Brunch/BIN?

People trying Brunch next here are not really clear about their own conditions, and I have no idea whether you can use Play Store or not. How about Linux apps?

I guess I need to try it myself to get answers...

**Comment by sebanc** [comment]
All I can confirm is that android apps are working for me on r85 builds using rammus recovery image.

The patch I removed in the latest r85 (in an attempt to fix android apps) might actually have broken things so I would suggest using the previous brunch r85 version to test android apps support.

Also, android won't work with slow usb flashdrives, so the dual boot method is recommended.

**Comment by bogdanenedoghitoiu** [comment]
Another update from me:
@qtotter linux apps are working fine.

I still couldn't get google play to work. I tried installing with the latest brunch r85 files and r84 r85 recovery images, with enable_updates and android_init_fix options and I couldn't connect to the play store.

Then I switched to the older brunch r85 files with rammus/hatch r84 r85 recovery images and enable_updates + android_init_fix options, still didn't work.

The r83 brunch files couldn't install at all, it would get stuck just like with the older kernel.

**Comment by sebanc** [comment]
@bogdanenedoghitoiu Could you post your /var/log/messages file ?

**Comment by sebanc** [comment]
@bogdanenedoghitoiu I just noticed your device has dual nvidia gpu, right ? Could you try adding "module_blacklist=nouveau" to the kernel command line ?

**Comment by bogdanenedoghitoiu** [comment]
[messages.txt](https://github.com/sebanc/brunch/files/5259327/messages.txt)
sure @sebanc , here it is

**Comment by bogdanenedoghitoiu** [comment]
> @bogdanenedoghitoiu I just noticed your device has dual nvidia gpu, right ? Could you try adding "module_blacklist=nouveau" to the kernel command line ?

trying now

**Comment by sebanc** [comment]
Well it seems it is the android display manager that fails for some reason:
[ARC] Received crash notification for surfaceflinger[10472]

But there is not much more in the log so it will be difficult to figure it out....

**Comment by bogdanenedoghitoiu** [comment]
Alright, no worries. Thank you for your help :)

I will keep an eye on this post and try to install using future releases.

**Comment by karthik-panchap** [comment]
FWIW, I'm hitting this consistently on a Surface Pro 7. Happy to try any workarounds. Thanks!

**Comment by qtotter** [comment]
@sebanc
> All I can confirm is that android apps are working for me on r85 builds using rammus recovery image.
Which rammus recovery image are you using? r83, r84 or r85? What is your CPU that you are using to test it with? Thx.

@bogdanenedoghitoiu
> Another update from me: (testing done on Lenovo IdeaPad 5 14IIL05)
> @qtotter linux apps are working fine.
> 
> I still couldn't get google play to work. I tried installing with the latest brunch r85 files and r84 r85 recovery images, with enable_updates and android_init_fix options and I couldn't connect to the play store.
> 
> Then I switched to the older brunch r85 files with rammus/hatch r84 r85 recovery images and enable_updates + android_init_fix options, still didn't work.
> 
> The r83 brunch files couldn't install at all, it would get stuck just like with the older kernel.
I guess your PC seems close to mine. Good to know, at least, Linux apps work. Thx.

**Comment by sebanc** [comment]
What I don't understand is that android apps appeared to work previously on 10th gen and I don't see what changed that.

Android apps however still work on those builds with my Surface Go and users with Intel Gen >=5 and <= 9 devices.

I have uploaded a new build, could someone with 10th gen try it ?

**Comment by bogdanenedoghitoiu** [comment]
@sebanc with what recovery image would you recommend installing? I might try today if I find the time

**Comment by sebanc** [comment]
I would suggest "rammus" as that's the one I use. "hatch" might also be an interesting one but I have no evidence showing that it would be better.

**Comment by bogdanenedoghitoiu** [comment]
@sebanc I tried both rammus and hatch r85 and google play doesn't start. It loads for a few minutes and then it shows the "Something went wrong." message. I tried both when booting up the computer and by manually activating it from settings and accepting the agreements and so on.

The installation was from a ssd (single OS boot on the laptop)

**Comment by qtotter** [comment]
> It loads for a few minutes and then it shows the "Something went wrong." message. I tried both when booting up the computer and by manually activating it from settings and accepting the agreements and so on.
> 
> The installation was done on a ssd (single OS boot on the laptop)

Your IdeaPad 5 14IIL05 has not i7 but i5, right?

**Comment by bogdanenedoghitoiu** [comment]
@qtotter yes I have an I5 cpu

**Comment by qtotter** [comment]
> @qtotter yes I have an I5 cpu

Your environment is quite similar to my PC which comes soon... Maybe, I should install Linux and Anbox if I fail installing ChromeOS...

**Comment by loki692** [comment]
My chrome is is not booting  after creating chrome OS from linux mint for about 4 days, now my windows also uninstalled please help me

**Comment by loki692** [comment]
#My chrome is is not booting  after creating chrome OS from linux mint for about 4 days, now my windows also uninstalled please help me

**Comment by princepjamie06832** [comment]
It seems Android app won't work on 10th Gen. Only works with 5th-9th Gen on brunch-next.

**Comment by loki692** [comment]
At which date  r86  release

**Comment by qtotter** [comment]
i finally received my pc; Inspiron 14 5000 Series 2-in1 -5400 with Core i5-1035G1/RAM 24GB/SSD 512GB. the chipset is of intel, and it is very simple and standard. 

anyway, i installed chrome os on USB drive and also on SSD, but the result was the same. it started running, i saw some lines on screen, and then, it started rebooting into efi... i could not even take a screenshot or anything because it just happened in a couple of seconds.

**Comment by sebanc** [comment]
@qtotter it seems there is something wrong with your install as you should at least see the system rebuild, could you upload a short video of the boot process ?

@princepjamie06832 @bogdanenedoghitoiu @chenshen2830569316 only issues are reported here but r85 has been reported to work on 10th gen with android apps by other users, I am not sure yet why it is not working with your devices though.
Have you all tried reseting google sync ? In chromeos: Settings -> People -> Sync and Google service -> Review your synced data -> Reset sync

**Comment by bogdanenedoghitoiu** [comment]
@sebanc yes I did try the google sync and even the powerwash on both rammus and hatch recovery images. I looked on their chromium os devices list and there they have a lenovo laptop with 10th gen intel cpu using hatch with 4.19 kernel version. Then I tried reinstalling using the stable brunch files but it would get stuck again at the intel hda thing. The disable_intel_hda option didn't make any difference on that first boot, couldn't get it to work.

 I also tried switching to beta/dev channels, the play store wouldn't start. Sometimes it would appear as "active" in the settings panel just when I started the computer but after clicking on the Play Store icon it would just load for a while and then throw the "Something went wrong." error. 

Unfortunately I have no way of showing some logs because I switched to Windows for a while so I can use the laptop for work. I honestly don't really mind that the Play Store isn't working but I need it to authorize the Android Debugging Bridge so I can debug mobile apps. If you know of any way of activating the adb on Chrome OS without Play Store let me know because I'd love to try. I managed to activate it from settings but it also needs to be authorized from the Play Store settings so that the Chrome OS device can be used as a debugging device from within Android Studio.

**Comment by qtotter** [comment]
@sebanc

Here's the boot video, but it's so fast and you cannot see anything...

https://drive.google.com/file/d/1Zx1HgDbkxrVdLFOIg4j4-LuM_kNb3XvP/view?usp=sharing

**Comment by sebanc** [comment]
@bogdanenedoghitoiu I was not referring to the 10th gen intel cpu working with hatch and 4.19 kernel version, this one just works with kernel 4.19 as some 10th gen i3 have a 9th gen gpu.
However, Daniel from the telegram group confirms android apps work with kernel 5.4:
`I'm using the latest Brunch r85 k5.4 testing with Rammus r85 on an i7-1065G7 and there are no issues with android apps.`
I am just saying that the android apps issue your are facing with kernel 5.4 is not a general one and is therefore probably not related to the kernel configuration. However, it might still be related to specific kernel issues with your device.
Could you attach a dmesg to this issue ?

@qtotter Your device seems to kernel panic with kernel 5.4, could you press "e" when you see the grub screen, add "module_blacklist=intel_lpss_pci" on the kernel command line (after cros_debug) and then press F10 to boot ?

**Comment by qtotter** [comment]
@sebanc, 

I added module_blacklist=intel_lpss_pci in grub, but the result was the same...

**Comment by bogdanenedoghitoiu** [comment]
@sebanc sorry for the misunderstanding, what I wanted to say is that I tried with both the 5.4 and 4.19 kernels. Good to hear that android apps work for some people, if I manage to do a reinstall this weekend I will send you the logs. Thanks for all the help!

**Comment by sebanc** [comment]
@qtotter my first attempt was a common fix for kernel panics on 5.4 kernel as I did not manage to read correctly the text on the video.
I eventually managed to zoom on the text and it seems the kernel panic is due to the usb subsystem. Do you have a dock connected to your laptop ? Also, do you have usb mode options in the bios ?

**Comment by qtotter** [comment]
@sebanc,

Wow, you could read the texts!? I don't use a dock, and the USB is connected directly to the PC.

What kind of USB options in BIOS are you thinking of?

**Comment by sebanc** [comment]
@qtotter Some settings like this:
https://www.dell.com/support/article/fr-fr/sln301218/how-to-boot-to-an-external-device-using-usb-type-c-connection?lang=en

Or maybe others usb related.

**Comment by johnnylau34** [comment]
> Hi,
> 
> I have added everyone in this issue to a new private "brunch-next" branch and uploaded a 5.4 build which appears to boot (at least on my Surface Go).
> 
> https://github.com/sebanc/brunch-next/releases
> 
> As previously mentioned, it's difficult to find the time to maintain 2 different kernels so I cannot guarantee regular/timely updates.

I got a surface pro 7 (i7, 256GB, 16GB) too and wish to participate in the testing too. Can you add me to this testing branch too?
Thanks!

**Comment by qtotter** [comment]
@sebanc,

i took a slow-motion video this time. 
https://drive.google.com/file/d/1ODFdoUkwavFSkkvpVYTqwM2T4zogTslA/view?usp=sharing

as for BIOS setting, it is set as follows;
Fastboot: Thorough
USB Boot Support: Enable
UEFI Network Stack: Enable 
(no legacy support by BIOS, and no thunderbolt on board)

when i installed chrome os on my internal ssd, the result was the same. it just keeps rebooting quickly... i am not sure if it's because i am booting from USB...

thanks for your support.

**Comment by qtotter** [comment]
@sebanc,

i used a better camera to capture the moment just before it reboots... hope this will tell you what's wrong with my pc. this time, used brunch_r85_k5.4_20200922 + chromeos_13310.59.0_hatch_recovery_stable-channel_mp-v6.bin. 

thx!
![DSC_1575 MOV_snapshot_00 32 032](https://user-images.githubusercontent.com/67125089/94987295-53934a00-05a0-11eb-8c06-a92f8858ac87.jpg)

**Comment by sebanc** [comment]
@johnnylau34 You should have received the github invite

@qtotter Thanks for the new screenshot. I was able to find a few things on google but those issues could only be solved by a bios update, is there one available for your laptop ?

**Comment by sebanc** [comment]
@qtotter Could you try booting with "acpi=off" added to the kernel command line ? It will probably not boot but if it goes further in the process it is probably an ACPI issue

A bios update remains the best option if there is one

**Comment by qtotter** [comment]
@sebanc,

as for bios, there is the latest bios released in aug on the dell site, but the version is the same on my pc... i will reinstall windows to update bios, anyway.

**Comment by sebanc** [comment]
No, if the bios the same version it is not necessary.

Now that I think about it, you can also try switching SATA configuration from RAID to AHCI in the bios

**Comment by qtotter** [comment]
Sata config has been always ahci to make USB boot work.

**Comment by sebanc** [comment]
@qtotter sorry I am out of ideas right now. The chromeos kernel is based on the linux one with a few adjustments.

Could you try to boot different ubuntu 20.04 on your usb flashdrive and see if it boots ? I believe it has 20.04 kernel also

**Comment by qtotter** [comment]
@sebanc,

I can install and boot Ubuntu 20.04 on the same USB drive.

**Comment by sebanc** [comment]
Ok, that's interesting, have you tried a previous brunch 5.4 version ?

**Comment by qtotter** [comment]
I just tried 20200915, but the result was the same.

**Comment by johnnylau34** [comment]
> @johnnylau34 You should have received the github invite
> 
> @qtotter Thanks for the new screenshot. I was able to find a few things on google but those issues could only be solved by a bios update, is there one available for your laptop ?

@sebanc 
Thanks a lot for your prompt response!

Tested on my Surface pro 7 i7 using "Brunch r85 k5.4 testing 20200922" + Chrome OS r85 and surprised how well it works:
**What works fine:** Wifi, bluetooth, Android apps, battery indicator, sleep (haven't got a chance to check if it can wake up properly after sleeping for long time though), general UI responsiveness
**What does not work at all:** Surface Pen
**What works partially only:** touch-screen - there seems to be a serious lag and can't properly register when the finger leaves the screen. Seems to work with single finger touch only (learnt similar limitation with jakeday linux kernel).

To me, the core theme of this issue "[Intel 10th Gen - Intel Iris Plus Graphics] Chrome OS not booting" is resolved for Surface Pro 7 i7 with Iris graphics. Should I raise a separate issue via brunch-next channel for the touch-screen issue instead?

**Comment by bogdanenedoghitoiu** [comment]
@sebanc do you think the country selection of the google account can have anything to do with the play store issue? I tried searching for an official retailer of chromebooks or pixelbooks in my country and couldn't find one.

**Comment by sebanc** [comment]
@johnnylau34 If I am not mistaken, only either "single touch" or "stylus" mode are supported for SP7 in the linux kernel (to enable stylus support instead of singletouch, add "ipts.singletouch=false" on the kernel command line). I think the linux-surface team is working on it, you can check their github.

@bogdanenedoghitoiu I am not aware of a specific issue related to the country selection but indeed you can try logging in with a US session.

**Comment by qtotter** [comment]
@sebanc, 

Just in case, I tried k4.19_stable_20200708 + rammus R53 again with slow-motion video camera. The last line says "ACPI memory or..." This laptop has 24GB RAM (I've added 16GB to original 8GB), and I wondered if this has anything to do with this kernel panic...?

![Screenshot_20201012-144621](https://user-images.githubusercontent.com/67125089/95710585-ef594000-0c9b-11eb-8bca-45d6fa7edaa3.png)

**Comment by sebanc** [comment]
@qtotter I believe the issue is with "acpi_ds_evaluate_name_path" function which means that kernel 4.19 did not manage to read correctly your device ACPI table. Even if it did, the kernel 4.19 would fail a bit later due to the lack of gpu driver for 10th gen. So, only kernel 5.4 has a chance to boot.

Regarding added ram, I don't think it could cause such an issue.

Once r86 lands on stable branch, I will release a new brunch 5.4 with a more recent kernel version which might potentially help with your issue.

**Comment by qtotter** [comment]
@sebanc,

As for graphics, my PC (CPU) has Intel UHD graphics, not Iris. Is UHD supported in the kernel?

**Comment by sebanc** [comment]
@qtotter I think both UHD and IRIS graphics are not supported for 10th gen on 4.19 kernel. Anyway the ACPI issue is also serious.

However, you can try brunch-next r86, we never know

**Comment by johnnylau34** [comment]
> @johnnylau34 If I am not mistaken, only either "single touch" or "stylus" mode are supported for SP7 in the linux kernel (to enable stylus support instead of singletouch, add "ipts.singletouch=false" on the kernel command line). I think the linux-surface team is working on it, you can check their github.
> 
> @bogdanenedoghitoiu I am not aware of a specific issue related to the country selection but indeed you can try logging in with a US session.

@sebanc I tried adding "ipts.singletouch=false" or "ipts.singletouch=true" in boot options but interestingly neither stylus nor single touch will work no matter which option is used; without specifying such option though, i can get single touch partially working.

**Comment by Bagronk** [comment]
Ok, so brunch next. What do I need to pull this off and this is a desktop so I will need to blacklist my nvidia GPU, if possible.

**Comment by Alexfebbraio** [comment]
I have a lenovo flex 5 15 2in1 i3 10gen with intel iris graphic.
i downloaded the latest brunch 86 stable with rammus 86 but i have the same problems as above. I also entered module_blacklist = intel_ishtp_hid options = iwlwifi_backport
but the problem remains the same

**Comment by sebanc** [comment]
@Alexfebbraio Validate the github invite you have received and try with "brunch-next" repository

**Comment by Alexfebbraio** [comment]
> @Alexfebbraio Validate the github invite you have received and try with "brunch-next" repository

Thank you sebanc. Now it's boot :-)
There is another problem i see network wifi but don't match password. I always get back wrong password

**Comment by qtotter** [comment]
R86 image and Brunch-next R86 didn't work on my laptop as before... A few sec of boot lines following rebooting.

I guess it's not a ChromeOS image problem...

**Comment by qtotter** [comment]
I don't think my Dell laptop is very different from other PCs with Intel 10th gen CPUs... Am I doing something wrong in the install process? Am I missing something??

**Comment by Alexfebbraio** [comment]
> > @Alexfebbraio Validate the github invite you have received and try with "brunch-next" repository
> 
> Thank you sebanc. Now it's boot :-)
> There is another problem i see network wifi but don't match password. I always get back wrong password

with nami 86 wifi works well. But the touchscreen doesn't work. Maybe it was better if I bought a dell 7000 15 2in1 (on my other dell 7000 17 2in1 everything works except the microphone)

**Comment by qtotter** [comment]
@sebanc,

i guess it's my bios settings... i'd appreciate if you could take a look at them when you have some spare time. Thanks in advance. 
https://drive.google.com/drive/folders/1mUGu9Nsmne0WW-d4hrPlV6CfNOJSYDqQ?usp=sharing

**Comment by sebanc** [comment]
@qtotter I looked at your bios config and it seems perfectly setup. Also, if your laptop is able to boot linux fine, it should also boot chromeos as it is globally the same kernel.

Google applies specific patches to the chromiumos kernel and my guess is that one of these specific patches causes a kernel panic on your laptop. However, identifying the culprit commit would be very difficult and I am not sure how to advance on this.

**Comment by Alexfebbraio** [comment]
> @Alexfebbraio Validate the github invite you have received and try with "brunch-next" repository

I tried with rammus, nami, eve and octupus. Same results:
No touchscreen
Yes accelerometer
Yes crostini
No Playstore (even with options = android_init_fix "

**Comment by sebanc** [comment]
@Alexfebbraio If you have a nvidia gpu, you can try to add "module_blacklist=nouveau" or "fbcon=map:1" on the kernel command line (after cros_debug)

If you don't have an nvidia gpu, the information I collected so far on the android issue that some 10th gen users are facing seems to indicate that some 10th gen cpu/gpu combinations are not correctly supported by the android container (SurfaceFlinger crashes) and unfortunately there is not much I can do about that.

**Comment by Alexfebbraio** [comment]
> @Alexfebbraio If you have a nvidia gpu, you can try to add "module_blacklist=nouveau" or "fbcon=map:1" on the kernel command line (after cros_debug)
> 
> If you don't have an nvidia gpu, the information I collected so far on the android issue that some 10th gen users are facing seems to indicate that some 10th gen cpu/gpu combinations are not correctly supported by the android container (SurfaceFlinger crashes) and unfortunately there is not much I can do about that.

Only iris graphics. 
the problem is with the wacom drivers different from those used for example lenovo flex 5 14 amd
I was unable to enable the touchscreen even with fedora 33 and ubuntu 20.10 which is possible with the dell 7000 17 2in1
Only with windows 10 grrrr

**Comment by qtotter** [comment]
@sebanc,

> @qtotter I looked at your bios config and it seems perfectly setup. 

Thanks for taking the time.

**Comment by DavidIlie** [comment]
Hello, it's been a while since I posted here. I just wanted to say that brunch works great on my laptop but there has been some issues.

I updated to the newest brunch release and also updated to chromeOS 86. Everything works great apart from the Camera and Microphone. My laptop is a HP Spectre x360 with a i7 1065G7

Is there any way to resolve these problems?

**Comment by mittal002** [comment]
Hello, Can you please give me acces to brunch next for 10th gen laptops

**Comment by vaugerbird** [comment]
Not 100% related, but HP just today announced a new Chromebook with an Ice Lake processor, which could mean a possible recovery image with better support for android on Ice Lake/10th Gen.

**Comment by Josh-110** [comment]
Hello sebanc, I've been following this project ever since I heard about it. I've successfully installed Chrome OS with Brunch on all my devices, and I am a fan. I recently bought a Surface Pro 7 i5 with Intel Iris Plus. I troubleshooted and it led me to this conversation. I am more than happy to help you test the brunch-next release, just let me know and please stay safe!

**Comment by sebanc** [comment]
@Josh-110 The github invite to "brunch-next" has been sent.

Actually, I profit of this occasion to correct a comment I made previously, I just re-read the linux-surface github and stylus does not currently work on SP7, only single touch is available.

**Comment by Josh-110** [comment]
> @Josh-110 The github invite to "brunch-next" has been sent.
> 
> Actually, I profit of this occasion to correct a comment I made previously, I just re-read the linux-surface github and stylus does not currently work on SP7, only single touch is available.

@sebanc thanks for sharing it! I already installed and I'm now testing it. It's pretty stable. As some users mentioned before, the touch barely works and camera definitely does not work. I also have a Surface Go and everything works except for camera and Youtube app which never loads. Youtube app does not with eitherSP7, It might be an issue with compatibly  but I'm not sure why. Thanks for taking your time to read this!

**Comment by dabinn** [comment]
@sebanc  Hi, I have same issue with Acer Spin 5 (SP513-54N). It is 10th gen Intel® Core™ i5-1035G4 / Iris® Plus Graphics.
Also stuck at  "input: HDA Intel PCH HDMI/DP, pcm=3" while booting.
Would you please join me to the "brunch-next" brunch?
![IMAG3855](https://user-images.githubusercontent.com/21699064/99296100-3204ce00-2881-11eb-884f-e0d05bb2d16e.jpg)

**Comment by mittal002** [comment]
@dabinn this is a different issue. Try turning of the os installed in it completely

**Comment by mittal002** [comment]
@dabinn I tested out several stable brunch-next reps and I put them on github.com/mittal002/brunch-next-gen. These are stable and public. **A BIG NOTE. THESE BUILDS ON THIS ARE BY SEBANC**

**Comment by dabinn** [comment]
@mittal002  Thanks, but I am not quite sure what you mean, Turn off the OS?

**Comment by mittal002** [comment]
@dabinn What I mean is for example if you have windows 10 or linux mint etc. then shut the down COMPLETELY. After that you can use the builds by sebanc over here github.com/mittal002/brunch-next-gen

**Comment by sebanc** [comment]
@dabinn The github invite for brunch-next has been sent. As you are suspecting, 5.4 kernel should fix your boot issue.

@mittal002 I am not sure why you think it is ok to make public mirrors of a private repo you have been invited to. Those releases are neither stable nor public so I would appreciate that you do not share them.

**Comment by dabinn** [comment]
@sebanc Thank you! I'll see if there anything I could help.

**Comment by mittal002** [comment]
@sebanc Sry I made it private again. Sry

**Comment by skkuweb** [comment]
Hello. Can you include me in the brunch-next brand? I have a Lenovo Ideapad3 15IIL laptop with i5-1035G4, but it's not booting again. I'm sorry if it's rude.

**Comment by sebanc** [comment]
@skkuweb Sure, no worries. The github email invite has been sent.

**Comment by Alexfebbraio** [comment]
> Hello. Can you include me in the brunch-next brand? I have a Lenovo Ideapad3 15IIL laptop with i5-1035G4, but it's not booting again. I'm sorry if it's rude.

Hello. i have the same laptop (i3 10gen). With brunch-next it starts but don't work playstore and touchscreen

**Comment by skkuweb** [comment]
@sebanc Thank you for your kindness. But I'm really sorry, but could you send invitation mail again? I didn't get an invitation mail. I'm sorry for the inconvenience.

**Comment by ryansg8** [comment]
@sebanc  Sorry for the spam you must get, but I am in the process of trying to install this onto my Ice Lake laptop, and access to brunch-next branch for kernel 5.4 would be awesome so I can get it booting. Thanks!

**Comment by sebanc** [comment]
@ryansg8 Github email invite sent.

**Comment by dabinn** [comment]
> @dabinn The github invite for brunch-next has been sent. As you are suspecting, 5.4 kernel should fix your boot issue.
> 
> @mittal002 I am not sure why you think it is ok to make public mirrors of a private repo you have been invited to. Those releases are neither stable nor public so I would appreciate that you do not share them.

I have tried the brunch-next and every thing works fine, thank you sebanc !
I'll post a full report later :D

**Comment by dabinn** [comment]
> > Hello. Can you include me in the brunch-next brand? I have a Lenovo Ideapad3 15IIL laptop with i5-1035G4, but it's not booting again. I'm sorry if it's rude.
> 
> Hello. i have the same laptop (i3 10gen). With brunch-next it starts but don't work playstore and touchscreen

@Alexfebbraio For the touchscreen, try to add "i8042.nopnp=1 pci=nocrs" to kernel parameters in Grub config.

**Comment by Alexfebbraio** [comment]
> > > Hello. Can you include me in the brunch-next brand? I have a Lenovo Ideapad3 15IIL laptop with i5-1035G4, but it's not booting again. I'm sorry if it's rude.
> > 
> > 
> > Hello. i have the same laptop (i3 10gen). With brunch-next it starts but don't work playstore and touchscreen
> 
> @Alexfebbraio For the touchscreen, try to add "i8042.nopnp=1 pci=nocrs" to kernel parameters in Grub config.

I entered the parameters in grub but the touchscreen still does not work :-(I have not even succeeded with fedora and ubuntu that I usually use. With my 2 DELL the touchscreen works.
Still don't understand what component it is. 
It's not wacom, it's not Elan. No information with dmesg, lspci, lsusb, hwinfo, xinput etc etc

**Comment by Alexfebbraio** [comment]
This is my hardware

[https://linux-hardware.org/?probe=4244d265ff](url)

**Comment by Alexfebbraio** [comment]
[https://linux-hardware.org/?probe=4244d265ff](url)

**Comment by us41242** [comment]
I was hoping to try out brunch-next on my SP7 if possible.

**Comment by sebanc** [comment]
@us41242 Sure, you should have received the github email invite.

**Comment by Josh-110** [comment]
hey Sebanc it's me again, just reporting on what i think it's a bit interesting. I have a portable external monitor powered by usb-c, and it has a touch screen. I was a bit surprised the touch input worked just fine and without any issues, 10 point touch supported. as I mentioned before, I am testing the brunch-next on a Surface Pro 7. I am testing it with the Chrome OS Atlas(Google Pixelbook Go) recovery image.

**Comment by Aster-the-Med-Stu** [comment]
Hello there! May I request for bruch-next repo access? I am using a Samsung Galaxy Book Flex which has 10th gen Intel i5. Thanks in advance!

**Comment by sebanc** [comment]
@Aster-the-Med-Stu You should have received the github email invite

**Comment by chenshen2830569316** [comment]
Today,I tried install chrome os on my hard disk,but still can’t use google play store,I think the reason is that I can’t reach setting complete interface，if possible,could you tell me what mirror do people use on surface pro 7,thank you.

**Comment by mithickartik** [comment]
Hello there, may I request the access for bruch-next access to install it on a 10th gen? I am have  Lenovo s340 which is a 10th gen Intel i3 laptop. Thankyou in advance!

**Comment by sebanc** [comment]
@mithickartik Actually I should have closed this issue, the latest stable release now has 5.4 kernel and is compatible with intel 10th gen.

Note: the android container is not compatible with some 10th gen.

**Comment by bybenjamin** [comment]
I have installed chrome os with brunch on another computer many times before. Fydeos works fine.
I did a single install without any problems. But brunch doesn't open.
Recovery: lantis 100 (I also tried rammus)
Brunch version: Brunch 100
Device features
CPU: N4500
Graphics Card: Intel UHD
Ram: 4GB
I am sharing the verbose output. Help me @sebanc
![Ekran görüntüsü 2022-05-19 101617](https://user-images.githubusercontent.com/91468765/169337951-2dc85f4a-1af8-4827-8adb-df1d6f919d75.jpg)
